lib/livebook/web_socket/client.ex

```diff
@@ -158,34 +143,23 @@ defmodule Livebook.WebSocket.Client do
     end
   end
 
-  defp handle_disconnect(conn, websocket, ref, result) do
-    with {:ok, conn, websocket} <- disconnect(conn, websocket, ref) do
-      {:ok, conn, websocket, result}
-    end
-  end
-
-  defp decode_response(websocket, %Response{status: 101, body: nil}) do
+  defp decode_response(websocket, %Response{status: 101, body: []}) do
     {websocket, {:ok, :connected}}
   end
 
   defp decode_response(websocket, response) do
     case Mint.WebSocket.decode(websocket, response.body) do
       {:ok, websocket, frames} ->
-        {websocket, handle_frames(response, frames)}
+        {websocket, {:ok, handle_frames(response, frames)}}
 
       {:error, websocket, reason} ->
         {websocket, {:error, reason}}
     end
   end
 
   defp handle_frames(response, frames) do
-    Enum.reduce(frames, response, fn
-      {:binary, binary}, acc ->
-        {:ok, %{acc | body: binary}}
-
-      {:close, _code, _data}, acc ->
-        {:close, acc}
-    end)
+    body = for {:binary, binary} <- frames, do: binary
```

Comment:
This is on purpose ignoring close events. Shouldn't we still match on close events and call `handle_disconnect`?